src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -73,21 +84,61 @@ internal class OnboardingSupervisor(
 
     private fun retrieveAssetsFromRouter() {
         if (StatsigConfig.useSkip) {
-            retrieveSkipTransferChains()
+            MainScope().launch {
+                retrieveSkipTransferChains()
+            }
             retrieveSkipTransferTokens()
         } else {
             retrieveTransferAssets()
         }
         retrieveCctpChainIds()
     }
 
-    private fun retrieveSkipTransferChains() {
+
+
+    private suspend fun retrieveSkipTransferChains() = coroutineScope {
```

Comment:
did compilation fail without `coroutineScope{ }` - i don't think this should be necessary since we're already launching in a scope.

Reply:
![Screenshot 2024-07-12 at 10 49 16 AM](https://github.com/user-attachments/assets/bb667392-ed84-4390-b177-719b0ffe2322)
this is the result w/o

Reply:
i believe the `coroutineScope` bit establishes scope for the async functions, otherwise, I'd have to do like `CoroutineScope(Dispatchers.Unconfined).async { ...` again

the documentation says:
> Creates a CoroutineScope and calls the specified suspend block with this scope. The provided scope inherits its coroutineContext from the outer scope, but overrides the context's Job.